### PR TITLE
Correctly handle the lifecycle of graph

### LINKED ISSUE #1394: Double check the lifecycle of context.
Ensure the lifecycle of context is correct to prevent memory-leak or invalid access.

- [x] The context must hold the graph even if the graph got unloaded.
- [x] The context memory can be freed by hand or when the context gets deleted.
- [x] If a graph is only held by a context, the unloading of the context should also unload the graph.

### PR DESCRIPTION
**Pull request by siyuan0322**
- Use del rather than unload
- Projected graph now hold a reference to the base graph

Fixes #1394

### CODE REVIEW

**Review by lidongze0629 — APPROVED** [review]

File: python/graphscope/framework/graph.py [hunk 1]
```diff
@@ -77,10 +77,6 @@ def add_edges(
     ):
         raise NotImplementedError
 
-    @abstractmethod
-    def unload(self):
-        raise NotImplementedError
```
Comment (sighingnow) [inline python/graphscope/framework/graph.py:82]: Please keep the `unload()` API, and raising a warning to tell users about the deprecation, like

```
warnings.warn("The Graph.unload() method has been deprecated, please using the `del` operator instead, e.g., `del graph`", DeprecationWarning)
```

**Review by sighingnow — CHANGES_REQUESTED** [review]
And, I suggest using `del` instead of private `_unload()` in affected test cases.

**Comment by siyuan0322** [comment]
> And, I suggest using `del` instead of private `_unload()` in affected test cases.

These remaining `_unload()` is for lazy... you must have a return value for session to run...

**Comment by sighingnow** [comment]
> > And, I suggest using `del` instead of private `_unload()` in affected test cases.
> 
> These remaining `_unload()` is for lazy... you must have a return value for session to run...

We could `sess.eval(g._unload())` in `_del` in both cases.

**Review by sighingnow — APPROVED** [review]

**Pull request merged**